development/benchmark-pr-comment.ts

```diff
@@ -0,0 +1,607 @@
+
+    for (const increase of significantIncreases) {
+      const currentFormatted = formatTime(increase.current);
+      const referenceFormatted = formatTime(increase.reference);
+      comment += `- **${increase.page} - ${increase.metric}**: ${currentFormatted} vs ${referenceFormatted} (**+${increase.percentIncrease.toFixed(1)}%**)\n`;
+    }
+
+    comment += `\n`;
+  }
+
+  comment += `---\n\n`;
+  comment += `*Results generated automatically by MetaMask CI*`;
+
+  return comment;
+}
+
+/**
+ * Main function that orchestrates the benchmark comment posting process.
+ * Reads benchmark results from file, generates a formatted comment, and posts it to the pull request.
+ *
+ * Required environment variables:
+ * - PR_COMMENT_TOKEN: GitHub token with permission to comment on PRs
+ * - OWNER: Repository owner (ex.: "MetaMask")
+ * - REPOSITORY: Repository name (ex.: "metamask-extension")
+ * - PR_NUMBER: Pull request number to comment on
+ *
+ * @throws {Error} When GitHub API request fails or required environment variables are missing
+ */
+async function main(): Promise<void> {
+  const { PR_COMMENT_TOKEN, OWNER, REPOSITORY, PR_NUMBER } =
+    process.env as Record<string, string>;
+  const N_COMMITS = 10;
+
+  if (!PR_NUMBER) {
+    console.warn('No pull request detected, skipping benchmark comment');
+    return;
+  }
+
+  if (!PR_COMMENT_TOKEN) {
+    console.warn('PR_COMMENT_TOKEN not provided, skipping benchmark comment');
+    return;
+  }
+
+  const filePath = path.join(
+    process.cwd(),
+    'test-artifacts/benchmarks/benchmark-results.json',
+  );
+
+  let benchmarkData: BenchmarkOutput | null = null;
+
+  try {
+    const fileContent = await fs.readFile(filePath, 'utf8');
+    benchmarkData = JSON.parse(fileContent) as BenchmarkOutput;
+    console.log(`Found benchmark results at: ${filePath}`);
+  } catch (error) {
+    console.log(`Could not read benchmark results from: ${filePath}`);
+  }
+
+  if (!benchmarkData) {
+    console.warn('No benchmark results found, skipping comment');
+    return;
+  }
+
+  const referenceData = await fetchLatestMainBenchmarkData(N_COMMITS);
+
+  const commentBody = generateBenchmarkComment(benchmarkData, referenceData);
+  console.log('Generated comment:');
+  console.log(commentBody);
+
+  // Post comment to PR
+  const JSON_PAYLOAD = JSON.stringify({ body: commentBody });
+  const POST_COMMENT_URI = `https://api.github.com/repos/${OWNER}/${REPOSITORY}/issues/${PR_NUMBER}/comments`;
+
+  console.log(`Posting to: ${POST_COMMENT_URI}`);
+
+  const response = await fetch(POST_COMMENT_URI, {
+    method: 'POST',
+    body: JSON_PAYLOAD,
+    headers: {
+      'User-Agent': 'metamaskbot',
+      Authorization: `token ${PR_COMMENT_TOKEN}`,
+      'Content-Type': 'application/json',
+    },
+  });
+
+  if (!response.ok) {
+    const errorText = await response.text();
+    throw new Error(
+      `Post comment failed with status '${response.statusText}': ${errorText}`,
+    );
+  }
```

Comment:
non blocking: this piece is exactly the same as the logic we have for the metamask bot, might we consider a function for both?

https://github.com/MetaMask/metamask-extension/blob/main/development/metamaskbot-build-announce.ts#L384